cmd/soroban-cli/src/config/sign_with.rs

```diff
@@ -88,6 +99,23 @@ impl Args {
         Ok(sign_txn_env(signer, tx_env, &network).await?)
     }
 
+    pub fn sign_tx_env_with_lab(&self, tx_env: &TransactionEnvelope) -> Result<(), Error> {
+        let passphrase = self.get_network()?.network_passphrase;
+        let xdr_buffer = tx_env.to_xdr_base64(Limits::none())?;
+
+        let mut url = Url::parse(&self.lab_url)?;
+        url.query_pairs_mut()
+            .append_pair("networkPassphrase", &passphrase)
+            .append_pair("xdr", &xdr_buffer);
+
+        let txn_sign_url = url.to_string();
+
+        println!("Opening lab to sign transaction: {}", &txn_sign_url);
```

Comment:
We have `--yes` flag that we can use here. 
I think we should add a prompt like "Lab link would be opened in the browser, do you want to continue? [Y/n]" 
And if `--yes` passed, this prompt should be skipped

Reply:
This makes sense to me!

I'm having trouble implementing it, and could use another set of eyes. The issue is that when I am piping my tx xdr into the cli, the program seems to lose track of the stdin reader when it gets to trying to read the user's `y` response.

I.e. when I pass in this command:
```
echo AAAAAgAAAADcOHnq5sGLOngOCEMyLqqn5CvFV2HGbOSjJAIzhqBdkAAAAGQAEb7UAAAAAQAAAAAAAAAAAAAAAQAAAAAAAAAYAAAAAQAAAAAAAAAAAAAAANw4eermwYs6eA4IQzIuqqfkK8VXYcZs5KMkAjOGoF2Q04SY/PMseMCgfbEFkX5gTr9qD/d0mvD9H3acFOz/X5oAAAAALtGgcgsAibk5q8VOLY2R1G1NdGX7fLzlz7iBFBy64JIAAAAAAAAAAAAAAAA= \
| cargo run tx sign --network testnet --sign-with-lab
```
I see this issue:
```
thread 'main' panicked at cmd/soroban-cli/src/config/sign_with.rs:150:41:
called `Result::unwrap()` on an `Err` value: Custom { kind: Other, error: "Failed to initialize input reader" }
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

It opens lab as expected with `--yes`.